README
Tk-Dressing

Tk::Dressing allows you to set a theme (a dressing) in your widget by using one of different default themes or by loading a new theme. 


How to participate to module to increase number of default themes ?

The first aim of Tk::Dressing is to allow user to choose between the proposed theme. But it is important 
for Tk::Dressing to propose a lot of default themes then, your help is welcome. 

Send me a .ini file and the name of your theme by email or through the web 
interface at : L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Tk-Dressing>. 
I will add it in the module in a new release. It will be notified in Change file.


INSTALLATION

To install this module, run the following commands:

	perl Build.PL
	./Build
	./Build test
	./Build install

Alternatively, to install with ExtUtils::MakeMaker, you can use the following commands:

	perl Makefile.PL
	make
	make test
	make install

SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the
perldoc command.

    perldoc Tk::Dressing

You can also look for information at:

    RT, CPAN's request tracker
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Tk-Dressing

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/Tk-Dressing

    CPAN Ratings
        http://cpanratings.perl.org/d/Tk-Dressing

    Search CPAN
        http://search.cpan.org/dist/Tk-Dressing/


LICENSE AND COPYRIGHT

Copyright (C) 2010 Djibril Ousmanou

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.



Changes
Revision history for Tk-Dressing

1.03    15/12/2010 21:04:28
        Themes updated because some color options had been forgotten for standard widgets.
        
1.02    11/12/2010 19:53:06
        minor bug resolved.
        POD documentation updated
        Thank you B. Jesson for yours new themes proposed
          - pink
          - purple
          - sunflower

1.01    10/12/2010 08:59:18
        POD documentation updated and test script corected.

1.00    09/12/2010 23:08:56
        First version of Tk::Dressing.



demo/test.pl
#!/usr/bin/perl
#==================================================================
# Author    : Djibril Ousmanou
# Copyright : 2010
# Update    : 11/12/2010 19:43:33
# AIM       : Test Tk::Dressing with standard widget
#==================================================================
BEGIN {
  my $error;
  foreach my $module (qw/ Tk::ColoredButton Tk::Canvas::GradientColor /) {
    eval "use $module;";
    $error .= "\t- $module\n" if ( $@ !~ /^\s*$/ );
  }
  if ( defined $error ) {
    warn("[ERROR] To use this test script you need to install theses modules :\n$error");
    exit;
  }
}

use warnings;
use Carp;
use strict;

use Tk;
use Tk::BrowseEntry;
use Tk::Canvas::GradientColor;
use Tk::ColoredButton;
use Tk::Dialog;
use Tk::DialogBox;
use Tk::DirTree;
use Tk::HList;
use Tk::LabFrame;
use Tk::MsgBox;
use Tk::NoteBook;
use Tk::Tree;
use Tk::Pane;
use Tk::ProgressBar;
use Tk::Table;
use Tk::TList;

use Tk::Dressing;

my $TkDressing = new Tk::Dressing;

my $mw = new MainWindow( -title => 'Using Tk::Dressing');
$mw->minsize( 400, 400 );

# menubar
my $BarMenu = $mw->Menu( -type => "menubar", );
$mw->configure( -menu => $BarMenu, );

my $FileMenu = $BarMenu->cascade( -label => 'Files', -tearoff => 0, );
$FileMenu->command( -label => 'File 1', );
$FileMenu->command( -label => 'File 2', -compound => "left" );
$FileMenu->command( -label => 'File 3', );
my $ExitMenu = $BarMenu->cascade( -label => 'exit', -tearoff => 0, );
$ExitMenu->command( -label => 'Close', -command => sub { exit; } );

my $label = $mw->Label( -text => "List of different standard widget" );
my $BrowseEntryTheme = $mw->BrowseEntry(
  -label   => "Theme : ",
  -state   => 'readonly',
  -choices => [ 'clear dressing', sort $TkDressing->get_all_theme ],
);
$BrowseEntryTheme->configure(
  -browse2cmd => sub {
    my $theme = $BrowseEntryTheme->Subwidget('entry')->get;
    if ( $theme eq 'clear dressing' ) { $TkDressing->clear($mw); return; }
    $TkDressing->design_widget(
      -widget => $mw,
      -theme  => $theme,
    );
  },
);

my $labframe = $mw->LabFrame( -label => 'My LabFrame' );

my $labelentry    = $labframe->Label( -text => "Entry" );
my $entry         = $labframe->Entry();
my $entrydisabled = $labframe->Entry( -text => 'Entry disabled', -state => 'disabled' );
my $but1          = $labframe->Button( -text => 'Test button' );
my $but2          = $labframe->Button( -text => 'Test button disabled', -state => 'disabled' );
my $ColoredButton1
  = $labframe->ColoredButton( -text => 'ColoredButton1', -tooltip => 'load file', -autofit => 1 );

my $checkbutton1 = $labframe->Checkbutton( -text => 'Example checkbutton', );
my $ColoredButton2 = $labframe->ColoredButton(
  -text    => 'CLEAR THEME',
  -tooltip => 'Button5',
  -font    => '{arial} 12 bold',
  ,
  -autofit  => 1,
  -gradient => { -start_color => '#FFCC33', -end_color => '#9999FF', },
  -command => sub { $TkDressing->clear($mw); return; },
);
my $ColoredButton3 = $labframe->ColoredButton(
  -text           => 'ColoredButton 3',
  -tooltip        => 'Button8',
  -gradient       => { -start_color => '#666666', -end_color => '#00B0D0' },
  -activegradient => { -start_color => '#60C000', -end_color => '#7000D0' },
);

# Radiobutton
my $Radiobutton1 = $labframe->Radiobutton( -text => 'Radiobutton1', );
my $Radiobutton2 = $labframe->Radiobutton( -text => 'Radiobutton2', -value => 0 );
my $Radiobutton3 = $labframe->Radiobutton( -text => 'Radiobutton3', -state => 'disabled' );

# messageBox, MsgBox, DialogBox and Dialog
my $but_messageBox = $labframe->Button(
  -text    => 'messageBox',
  -command => sub {
    my $reponse_messageBox = $labframe->messageBox(
      -icon    => 'info',
      -title   => 'Message',
      -type    => 'OK',
      -message => 'Tk::Dressing not work in messageBox',
    );
  }
);
my $but_MsgBox = $labframe->Button(
  -text    => 'MsgBox',
  -command => sub {
    my $MsgBox = $labframe->MsgBox(
      -title   => 'MsgBox',
      -type    => 'okcancel',
      -message => 'Tk::Dressing not work in MsgBox',
    );

    my $button = $MsgBox->Show;
  }
);
my $but_DialogBox = $labframe->Button(
  -text    => 'DialogBox',
  -command => sub {
    my $DialogBox = $labframe->DialogBox(
      -title   => 'DialogBox',
      -buttons => [ "OK", "Cancel" ]
    );
    $DialogBox->Label( -text => "Test DialogBox", )->pack();
    $TkDressing->design_widget(
      -widget => $DialogBox,
      -theme  => $BrowseEntryTheme->Subwidget('entry')->get,
    );

    $DialogBox->Show;
  }
);
my $but_Dialog = $labframe->Button(
  -text    => 'Dialog',
  -command => sub {
    my $dialog = $mw->Dialog(
      -text           => 'Save File?',
      -bitmap         => 'question',
      -title          => 'Save File Dialog',
      -default_button => 'Yes',
      -buttons        => [qw/Yes No Cancel/],
    );
    $TkDressing->design_widget(
      -widget => $dialog,
      -theme  => $BrowseEntryTheme->Subwidget('entry')->get,
    );
    $dialog->Show;
  }
);

# Scale
my $scale = $labframe->Scale(
  -label     => 'Scale example',
  -from      => 0,
  -to        => 100,
  -orient    => 'horizontal',
  -takefocus => 1
);

# Spinbox
my $Spinbox = $labframe->Spinbox( qw/-from 1 -to 10 -width 10 -validate key/, );

# Listbox
my $liste = $labframe->Listbox();
$liste->insert( 'end', qw/perl java R C++/ );

my $hlist = $labframe->HList(
  -itemtype   => 'text',
  -separator  => '/',
  -selectmode => 'single',
  -browsecmd  => sub {
    my $file = shift;
    $label->configure( -text => $file );
  }
);
foreach (qw(/ /home /home/ioi /home/foo /usr /usr/lib)) {
  $hlist->add( $_, -text => $_ );
}

my $tree = $labframe->Scrolled('Tree');
foreach (
  qw/orange orange.red orange.yellow green green.blue
  green.yellow purple purple.red purple.blue/
  )
{
  $tree->add( $_, -text => $_ );
}

my $labframeonglet = $mw->LabFrame( -label => 'LabFrame onglets' );

# NoteBook
my $NoteBook = $labframeonglet->NoteBook();
my $onglet1  = $NoteBook->add( "onglet 1", -label => "Tk::Text", );
my $onglet2  = $NoteBook->add( "onglet 2", -label => "Canvas", );
my $onglet3  = $NoteBook->add( "onglet 3", -label => "Menu", );
my $onglet4  = $NoteBook->add( "onglet 4", -label => "disabled", -state => 'disabled' );
my $onglet5  = $NoteBook->add( "onglet 5", -label => "Tk::Table" );
my $onglet6  = $NoteBook->add( "onglet 6", -label => "Tk::TList" );
my $onglet7  = $NoteBook->add( "onglet 7", -label => "Tk::ProgressBar" );

my $WidgetText = $onglet1->Scrolled( 'Text', -scrollbars => 'osoe', );
$WidgetText->insert( 'end', "1 : data example\n" );
$WidgetText->insert( 'end', "2 : data example\n" );

# Canvas GradientColor
my $Canvas = $onglet2->GradientColor();
my $but_canvas = $Canvas->Button( -text => 'Test button' );
$Canvas->createOval( 10, 10, 50, 50, -fill => 'green' );
$Canvas->createWindow( 75, 75, -window => $but_canvas );

my $Menubutton = $onglet3->Menubutton(
  -text      => 'Menubutton',
  -menuitems => [
    [ 'command', => 'one' ],
    [ 'command', => 'two' ],
    '-',
    [ 'command', => 'three' ],
    [ 'command', => 'four' ],
  ],
);
my $languagemenu = $Menubutton->cascade(
  -label   => 'Your language please ?',
  -tearoff => 0,
);
$languagemenu->checkbutton( -label => 'Perl', );
$languagemenu->checkbutton( -label => 'JAVA', );

my $Optionmenu = $onglet3->Optionmenu(
  -options => [
    [ 'January'   => 1 ],
    [ 'February'  => 2 ],
    [ 'March'     => 3 ],
    [ 'April'     => 4 ],
    [ 'May'       => 5 ],
    [ 'June'      => 6 ],
    [ 'July'      => 7 ],
    [ 'August'    => 8 ],
    [ 'September' => 9 ],
    [ 'October'   => 10 ],
  ],
);

# Tk::Table
my ( @cell_vars, $row, $col );
foreach $row ( 0 .. 9 ) {
  my @row_vars;
  foreach $col ( 0 .. 9 ) {
    push @row_vars, $row * $col;
  }
  push @cell_vars, \@row_vars;
}

my $table = $onglet5->Table(
  -rows         => 6,
  -columns      => 9,
  -scrollbars   => 'os',
  -fixedrows    => 1,
  -fixedcolumns => 1,
)->pack;

foreach $col ( 1 .. 9 ) {
  my $col_header = $onglet5->Button( -text => "Column " . $col );
  $table->put( 0, $col, $col_header );
}

foreach $row ( 1 .. 5 ) {
  my $row_header = $onglet5->Button( -text => "Row " . $row );
  $table->put( $row, 0, $row_header );
  foreach $col ( 1 .. 9 ) {
    my $cell = $onglet5->Entry( -width => 10, -textvariable => \$cell_vars[$row][$col] );
    $table->put( $row, $col, $cell );
  }
}

# TList
my $tlist = $onglet6->Scrolled( 'TList', -scrollbars => 'osoe', -width => 30, -height => 20, )->pack();
for ( my $i = 0; $i < 50; $i++ ) {
  $tlist->insert( $i, -itemtype => 'text', -text => "This is item $i" );
}

# ProgressBar
my $progress = $onglet7->ProgressBar(
  -from   => 0,
  -to     => 100,
  -length => 200,
  -colors => [ 0, 'red' ],
  -width  => 20,
)->pack(qw / -pady 10 /);
$onglet7->Button( -text => "start progressbar", -command => \&ProgressBar )->pack(qw / -pady 10 /);

# Display widget
$label->grid( $BrowseEntryTheme, qw/ -padx 10 -pady 10 / );
$labframe->grid( $labframeonglet, qw/ -padx 10 -pady 10 / );

$labelentry->grid( $entry, $entrydisabled, qw/ -padx 10 -pady 10 / );
$but1->grid( $but2, $ColoredButton1, qw/ -padx 10 -pady 10 / );

$NoteBook->pack(qw/ -expand 1 -fill both -padx 10 -pady 10 /);
$WidgetText->grid(qw/ -padx 10 -pady 10 /);
$Canvas->grid(qw/ -padx 10 -pady 10 /);
$Menubutton->grid( $Optionmenu, qw/ -padx 10 -pady 10 / );

$checkbutton1->grid( $ColoredButton2, $ColoredButton3, qw/ -padx 10 -pady 10 / );
$Radiobutton1->grid( $Radiobutton2,   $Radiobutton3,   qw/ -padx 10 -pady 10 / );
$but_messageBox->grid( $but_MsgBox, $but_DialogBox, qw/ -padx 10 -pady 10 / );
$but_Dialog->grid( $Spinbox, $scale, qw/ -padx 10 -pady 10 / );
$liste->grid( $hlist, $tree, qw/ -padx 5 -pady 5 / );

MainLoop;

sub ProgressBar {
  for ( my $i = 0; $i <= 100; $i = $i + 10 ) {
    $progress->value($i);
    $progress->update;
    sleep 1;
  }
}


lib/Tk/Dressing.pm
package Tk::Dressing;

use warnings;
use strict;
use Carp;

#==================================================================
# Author    : Djibril Ousmanou
# Copyright : 2010
# Update    : 15/12/2010 22:02:36
# AIM       : Set a design for a Tk widget and its children
#==================================================================

use Tk;
use Config::Std { def_gap => 0 };
use File::Basename qw/ dirname /;
use File::Copy qw / copy /;
use File::Spec;

use vars qw($VERSION);
$VERSION = '1.03';

# get theme directory
my $themes_directory = File::Spec->catfile( dirname( $INC{'Tk/Dressing.pm'} ), 'DressingThemes' );

my %new_theme;
my %initial_theme;
my $current_theme;

sub new {
  my ($self) = @_;

  $self = ref($self) || $self;
  my $this = {};
  bless( $this, $self );

  return $this;
}

sub get_current_theme {
  my $this = shift;

  return $current_theme;
}

sub get_all_theme {
  my $this = shift;

  opendir( THEME, $themes_directory ) or croak "Unable to read themes directory : $themes_directory\n";
  my @alltheme = grep {/.ini$/} readdir(THEME);
  foreach (@alltheme) {
    s{\.ini$}{};
  }
  closedir(THEME);

  # New themes loaded
  push( @alltheme, keys %new_theme );

  my %unique;
  @unique{@alltheme} = ();

  return keys %unique;
}

sub get_default_theme_file {
  my ( $this, $theme, $directory ) = @_;

  unless ( defined $theme ) {
    carp("Theme not defined\n");
    return;
  }

  $directory = defined $directory ? $directory : '.';

  my $theme_file     = "$themes_directory/$theme.ini";
  my $new_theme_file = "$directory/$theme.ini";

  # default theme file
  if ( -e $theme_file ) {
    copy( $theme_file, $new_theme_file );
    return $new_theme_file;
  }
  carp("$theme not found\n");

  return;
}

sub load_theme_file {
  my ( $this, $theme, $theme_file ) = @_;

  if ( -e $theme_file ) {
    read_config $theme_file => my %config;
    $new_theme{$theme} = \%config;
    return;
  }

  carp("$theme_file not found\n");
  return;
}

sub clear {
  my ( $this, $widget ) = @_;

  unless ( defined $widget and Exists $widget) {
    carp("Widget not defined\n");
    return;
  }

  if (%initial_theme) {
    $this->design_widget( -widget => $widget, -clear => 1 );
    $current_theme = undef;
  }

  return 1;
}

sub _default_config {
  my ( $this, $widget, $ref_config_theme ) = @_;

  my ( $class, $type ) = split( /::/, ref($widget) );
  $type = defined $type ? $type : $class;
  return unless defined $type;

  # Store the initial configuration before set the first theme
  unless ( $initial_theme{$type} ) {
    foreach my $option ( sort keys %{ $ref_config_theme->{$type} } ) {

      #my ( $name, undef, undef, $default_value, $current_value ) = $widget->configure($option);
      #my $ValueInitial = defined $current_value ? $current_value : $default_value;
      #print "$type => $name : $default_value, $current_value\n";

      my $ValueInitial = $widget->cget($option);
      next unless defined $ValueInitial;
      
      if ( $ValueInitial eq 'SystemButtonFace' and $type eq 'Entry' ) {
        $ValueInitial = 'white' if ( $option eq '-background' );
      }
      elsif ( $ValueInitial eq 'SystemWindow' and $type eq 'Frame' ) {
        $ValueInitial = 'SystemButtonFace';
      }
      $initial_theme{$type}{$option} = $ValueInitial;
    }
  }

  # children widgets design setting
  foreach my $child_level1 ( sort $widget->children ) {
    $this->_default_config( $child_level1, $ref_config_theme );
  }
  return;
}

#============================================
# set design in widget and its children
#============================================
sub design_widget {
  my ( $this, %information ) = @_;

  my $theme = $information{-theme} || 'djibel';
  unless ( defined $theme ) {
    carp("Theme not defined\n");
    return;
  }

  my $widget = $information{-widget};
  unless ( defined $widget and Exists $widget) {
    carp("Widget not defined\n");
    return;
  }

  my $clear = $information{-clear};

  # Get theme configuration
  my $ref_config_theme;

  # Clear
  if ( defined $clear and $clear == 1 ) {
    $ref_config_theme = \%initial_theme;
  }
  elsif ( exists $new_theme{$theme} ) {
    $ref_config_theme = $new_theme{$theme};
  }
  else {
    my $theme_file = "$themes_directory/$theme.ini";
    unless ( -e $theme_file ) {
      carp("$theme not found\n");
      return;
    }

    read_config $theme_file => my %config;
    $new_theme{$theme} = \%config;
    $ref_config_theme = $new_theme{$theme};
  }
  return unless defined $ref_config_theme;

  # Get Default configuration
  unless ( %initial_theme or ( defined $clear and $clear == 1 ) ) {
    $this->_default_config( $widget, $ref_config_theme );
  }

  # Get Class an type of widget to design it
  my ( $class, $type ) = split( /::/, ref($widget) );

  # For MainWindows widget, ref($widget) = MainWindow, then $type = $class
  # Else for Tk::Widget, ref($widget) = Tk::Toplevel or Tk::Frame, etc => $type ok
  $type = defined $type ? $type : $class;
  return unless defined $type;

  # Read configuration option
  if ( my $design_type = $ref_config_theme->{$type} ) {
    # Set configuration
    $widget->configure( %{$design_type} );
  }

  # children widgets design setting
  foreach my $child_level1 ( $widget->children ) {
    if ( defined $clear and $clear == 1 ) {
      $this->design_widget( -widget => $child_level1, -clear => 1 );
    }
    else {
      $this->design_widget( -widget => $child_level1, -theme => $theme );
    }
  }

  $current_theme = $theme;
  return;
}

1;
__END__

=head1 NAME

Tk::Dressing - Set a theme (dressing) in your widget and its children.

=head1 SYNOPSIS

  #!/usr/bin/perl
  use warnings;
  use strict;
  use Tk;
  use Tk::Dressing;
  
  use Tk::BrowseEntry;
  
  my $TkDressing = new Tk::Dressing;
  
  my $mw = new MainWindow( -title => "Dressing widget", );
  $mw->minsize( 300, 100 );
  
  $mw->Button( -text => 'Close', )->pack(qw/ -side bottom -padx 10 -pady 10 /);
  
  my $BrowseEntryTheme = $mw->BrowseEntry(
    -label   => "Select a theme : ",
    -state   => 'readonly',
    -choices => [ 'clear dressing', sort $TkDressing->get_all_theme ],
  )->pack;
  
  my $Message = "Hello everybody\n\nWelcome to Perl/Tk and Tk::Dressing\n\n";
  $mw->Label( -text => $Message,     -anchor => 'center' )->pack(qw/ -side top -padx 10 -pady 10 /);
  $mw->Label( -text => 'Example : ', -anchor => 'center' )->pack(qw/ -side left -padx 10 -pady 10 /);
  $mw->Entry( -text => 'test', )->pack(qw/ -side left -padx 10 -pady 10 /);
  
  $BrowseEntryTheme->configure(
    -browse2cmd => sub {
      my $theme = $BrowseEntryTheme->Subwidget('entry')->get;
      if ( $theme eq 'clear dressing' ) { $TkDressing->clear($mw); return; }
      $TkDressing->design_widget(
        -widget => $mw,
        -theme  => $theme,
      );
    },
  );
  
  MainLoop();


=head1 DESCRIPTION

Tk::Dressing allows you to set a theme (dressing) in your widget and its children by using one of different default 
themes or by loading a new theme. A theme contains all options that you want to use to configure the Tk widgets 
(color of buttons, of Entry, ...). 

Everybody can participate to increase the themes of this module by proposing a theme that 
will be store in the module because each theme is stored in an ini file.

=head1 CONSTRUCTOR/METHODS

=head2 new

This constructor allows you to create a new Tk::Dressing object.

B<$TkDressing = new Tk::Dressing>

The new() method is the main constructor for the Tk::Dressing module.

  # Create Tk::Dressing constructor
  my $TkDressing = new Tk::Dressing;


=head2 clear

This method allow you to replace your last theme set by the default look and feel of your widget.

B<$TkDressing-E<gt>clear( $Widget )>

  $TkDressing->clear($mw);


=head2 design_widget

Set the theme (dressing) on your widget and its children.

B<$TkDressing-E<gt>design_widget( -widget =E<gt> $Widget, -theme =E<gt> $ThemeName )>

=over 4

=item B<-widget> =E<gt> I<$Widget>

Specifies the widget. The widget can be a Toplevel, Frame, Button, ...

  -widget => $mw,

=back

=over 4

=item B<-theme> =E<gt> I<string>

Specifies the theme name between defaults themes of module. If you have loaded a theme before, you can choose it.

    -theme => 'starwars',

Default : B<djibel>

=back

  $TkDressing->design_widget(
   -widget => $mw,
   -theme  => 'djibel',
  );

=head2 get_all_theme

Get all the available themes and loaded in your widget.

  my @all_themes = $TkDressing->get_all_theme;


=head2 get_current_theme

Get the theme in used in your widget.

  my $current_theme = $TkDressing->get_current_theme;


=head2 get_default_theme_file

Get an .ini file of one of default themes of module.

B<$file> = B<$TkDressing-E<gt>get_default_theme_file(I<$theme>, I<$directory>)>

  my $file = $TkDressing->get_default_theme_file('djibel', '/home/user');
  # $file will be contain /home/user/djibel.ini and this file will be create.


=head2 load_theme_file

Loads your own theme file to design your widget.

  # Load your file
  $TkDressing->load_theme_file($theme, $theme_file);
  # Set it to e frame widget
  $TkDressing->design_widget(
   -widget => $Frame,
   -theme  => $theme,
  );


Your file must be an .ini file and must contain sections. 
Each section correspond to a widget, that is an example.

  [BrowseEntry]
  -background: #697E6D
  -foreground: white
  -disabledbackground: #697E6D
  -disabledforeground: white
  
  [Button]
  -activebackground: #7F80C0
  -background: #7F80C0
  -foreground: white
  -disabledforeground: white
  -activeforeground: white
  
  [Canvas]
  -background: #697E6D
  
  [Checkbutton]
  -highlightbackground: #68676C
  -activebackground: #68676C
  -background: #68676C
  -foreground: white
  -disabledforeground: white
  -activeforeground: white
  -selectcolor: #697E6D
  
  [ColoredButton]
  -highlightbackground: #68676C
  -background: #68676C
  -autofit: 1
  
  [DirTree]
  -highlightbackground: #68676C
  -background: #697E6D
  -foreground: white
  -highlightcolor: #68676C
  -selectforeground: white
  -selectbackground: #68676C
  
  [Entry]
  -highlightbackground: #68676C
  -background: #697E6D
  -disabledbackground: #697E6D
  -foreground: white
  -readonlybackground: #697E6D
  -disabledforeground: white
  -insertbackground: white

=head1 How to participate to module to increase number of default themes ?

The first aim of Tk::Dressing is to allow user to choose between the proposed theme. But it is important 
for Tk::Dressing to propose a lot of default themes then, your help is welcome. 

Send me a .ini file and the name of your theme by email or through the web 
interface at : L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Tk-Dressing>. 
I will add it in the module in a new release. It will be notified in Change file.

=head1 EXAMPLE

See the demo directory in the distribution to execute an example program using Tk::Dressing.

=head1 SEE ALSO

See L<Tk::CmdLine> and Tk::Preferences.

=head1 AUTHOR

Djibril Ousmanou, C<< <djibel at cpan.org> >>

=head1 BUGS

Please report any bugs or feature requests to C<bug-tk-dressing at rt.cpan.org>, or through
the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Tk-Dressing>.  I will be notified, and then you'll
automatically be notified of progress on your bug as I make changes.


=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc Tk::Dressing

You can also look for information at:

=over 4

=item * RT: CPAN's request tracker

L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Tk-Dressing>

=item * AnnoCPAN: Annotated CPAN documentation

L<http://annocpan.org/dist/Tk-Dressing>

=item * CPAN Ratings

L<http://cpanratings.perl.org/d/Tk-Dressing>

=item * Search CPAN

L<http://search.cpan.org/dist/Tk-Dressing/>

=back


=head1 ACKNOWLEDGEMENTS


=head1 LICENSE AND COPYRIGHT

Copyright 2010 Djibril Ousmanou.

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See http://dev.perl.org/licenses/ for more information.


=cut

